### ISSUE TITLE
Migrate plugins API on the new service hub API

### ISSUE THREAD

**Issue by abe33** [atom-minimap/minimap#267]
Probably something that will need a major bump (v5), but now that Atom provides a service API, plugins should be migrated to use this API. 
It should fix the issue of plugins not registered after a minimap update as it'll be the service hub that will deal with activation. 
I would like to keep the ability to quickly toggle a plugin for testing purpose from the quick setting view.

**Comment by fundon** [comment]
API link?

**Comment by abe33** [comment]
Sorry, I completely forgot to update this issue.
Here's the link: https://atom.io/docs/v0.184.0/creating-a-package#interacting-with-other-packages-via-services

Actually the change was rather simple and it coexist well with the current implementation so I shipped the service in version 4.4.0. The service just have to return the main object and plugins receive it in the consuming method to register. 
I updated `minimap-selection` to use the new service API, I think transitionning all the plugins can be done quite quickly I'll try to push their update at the start of the week.
